redfish-core/lib/processor.hpp

```diff
@@ -1080,7 +1080,8 @@ inline void
                 {
                     aResp->res.jsonValue["Status"]["State"] = "Absent";
                 }
-                else if (functional == false)
+
+                if (functional == false)
```

Comment:
I am not sure this will fix the SW545991 defect because the defect says the `Health` property value showed as expected but, the`State` property showed as `Enabled` when it is deallocated. Looks we are updating only Health property in this case?

Reply:
Still, I am not sure this will fix SW545991 but, it will fix SW546084 so, I will approve.